cuda_pathfinder/cuda/pathfinder/_headers/find_nvidia_headers.py

```diff
@@ -0,0 +1,44 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import functools
+import glob
+import os
+from typing import Optional
+
+from cuda.pathfinder._dynamic_libs.supported_nvidia_libs import IS_WINDOWS
+from cuda.pathfinder._utils.find_sub_dirs import find_sub_dirs_all_sitepackages
+
+
+@functools.cache
+def find_nvidia_header_directory(libname: str) -> Optional[str]:
+    if libname != "nvshmem":
+        raise RuntimeError(f"UNKNOWN {libname=}")
+
+    if libname == "nvshmem" and IS_WINDOWS:
+        # nvshmem has no Windows support.
+        return None
+
+    # Installed from a wheel
+    nvidia_sub_dirs = ("nvidia", "nvshmem", "include")
+    hdr_dir: str  # help mypy
+    for hdr_dir in find_sub_dirs_all_sitepackages(nvidia_sub_dirs):
```

Comment:
We should do it in a follow up, but I think we should really move towards using an `importlib` based resolution method instead of walking the sitepackages ourselves.

Reply:
> I thought we discussed and agreed walking the paths is acceptable since on the PyPI side they are predictable? Any reason to prefer importlib?

Lines crossed here. Let's review under #864 when we get to it.

Reply:
864?